File: flutter_app/lib/features/todo/data/remote/todo_dto.dart
import '../../todo.dart';

class TodoDto {
  const TodoDto({
    required this.id,
    required this.title,
    required this.completed,
  });

  final int id;
  final String title;
  final bool completed;

  factory TodoDto.fromJson(Map<String, dynamic> json) => TodoDto(
        id: json['id'] as int,
        title: json['title'] as String,
        completed: json['completed'] as bool,
      );

  Todo toEntity() => Todo(
        id: id.toString(),
        title: title,
        isCompleted: completed,
        createdAt: DateTime.fromMillisecondsSinceEpoch(0),
      );
}

File: android_xml/app/src/main/java/com/example/androidxml/features/todo/data/remote/TodoDto.kt
package com.example.androidxml.features.todo.data.remote

import com.example.androidxml.features.todo.Todo
import com.example.androidxml.features.todo.TodoPriority
import com.google.gson.annotations.SerializedName

data class TodoDto(
    @SerializedName("id") val id: Int,
    @SerializedName("title") val title: String,
    @SerializedName("completed") val completed: Boolean,
)

fun TodoDto.toEntity() = Todo(
    id = "api_$id", title = title, isCompleted = completed,
    description = "APIから取得", priority = TodoPriority.MEDIUM,
    createdAt = System.currentTimeMillis(),
)
